weld/subsystem/src/main/java/org/jboss/as/weld/WeldSubsystemAdd.java

```diff
@@ -67,6 +74,30 @@
 
     static final WeldSubsystemAdd INSTANCE = new WeldSubsystemAdd();
 
+    static {
```

Comment:
Can you elaborate why this logic is necessary in conjuction with this change?

Reply:
This extension uses CDI.current() instead of java:comp:BeanManager. CDI.current() presently just returns the root BM of the EAR, which does not have visibility to beans defined in a WAR. 

This change means that the extension will find the appropriate beans, even in the presence of  multiple WARs.

Really this could probably be considered a seperate bug that could be fixed independently.

Reply:
Probably it would be better to treat as separate WELD bug to have visibility/attention of weld experts.